ios/RNSScreenStack.m

```diff
@@ -557,6 +557,23 @@ - (void)dismissOnReload
   return nil;
 }
 
+- (BOOL)isInGestureResponseDistance:(UIGestureRecognizer *)gestureRecognizer topScreen:(RNSScreenView *)topScreen
+{
+  NSDictionary *gestureResponceDistanceValues = topScreen.gestureResponseDistance;
+  float x = [gestureRecognizer locationInView:gestureRecognizer.view].x;
+  float y = [gestureRecognizer locationInView:gestureRecognizer.view].y;
+  BOOL isRTL = _controller.view.semanticContentAttribute == UISemanticContentAttributeForceRightToLeft;
+  if (isRTL) {
+    x = _controller.view.frame.size.width - x;
+  }
+  // we check if any of the contstraints are violated and return NO if so
+  return !(
+      (gestureResponceDistanceValues[@"minX"] && x < [gestureResponceDistanceValues[@"minX"] floatValue]) ||
+      (gestureResponceDistanceValues[@"maxX"] && x > [gestureResponceDistanceValues[@"maxX"] floatValue]) ||
+      (gestureResponceDistanceValues[@"minY"] && y < [gestureResponceDistanceValues[@"minY"] floatValue]) ||
+      (gestureResponceDistanceValues[@"maxY"] && y > [gestureResponceDistanceValues[@"maxY"] floatValue]));
+}
```

Comment:
This implementation seems to be very limiting. For example I'd like to create a virtual square that only in these bounds would gesture recognizer work:

![image](https://user-images.githubusercontent.com/39658211/152358652-958af739-94a9-4376-8afe-1e5f1caab18f.png)




 but I don't think it's possible with this setup.

Reply:
Since all the limiting values are resolved from top left corner, you can achieve it with something like 
```js
{
	minX: 400,
	maxX: 500,
	minY: 200,
	maxY: 300,
}
```
does it answer the problem?